packages/matter.js/src/protocol/interaction/InteractionMessenger.ts

```diff
@@ -166,35 +166,67 @@ export class InteractionServerMessenger extends InteractionMessenger<MatterDevic
 
     async sendDataReport(dataReport: DataReport) {
         const messageBytes = TlvDataReport.encode(dataReport);
-        if (dataReport.attributeReports !== undefined && messageBytes.length > MAX_SPDU_LENGTH) {
+        if ((dataReport.attributeReports !== undefined || dataReport.eventReports !== undefined) && messageBytes.length > MAX_SPDU_LENGTH) {
             // DataReport is too long, it needs to be sent in chunks
-            const attributeReportsToSend = [...dataReport.attributeReports];
-            dataReport.attributeReports.length = 0;
+            const attributeReportsToSend = [...(dataReport.attributeReports ?? [])];
+            const eventReportsToSend = [...dataReport.eventReports ?? []];
+            dataReport.attributeReports = undefined;
+            dataReport.eventReports = undefined;
             dataReport.moreChunkedMessages = true;
 
             const emptyDataReportBytes = TlvDataReport.encode(dataReport);
 
             let messageSize = emptyDataReportBytes.length;
             while (true) {
-                const attributeReport = attributeReportsToSend.pop();
-                if (attributeReport === undefined) {
+                if (attributeReportsToSend.length > 0) {
+                    const attributeReport = attributeReportsToSend.shift();
+                    if (attributeReport !== undefined) {
+                        const attributeReportBytes = TlvAttributeReport.encode(attributeReport).length;
+                        if (messageSize + attributeReportBytes > MAX_SPDU_LENGTH) {
+                            if (messageSize === emptyDataReportBytes.length) {
+                                throw new NotImplementedError(`Attribute report for is too long to fit in a single chunk, Array chunking not yet supported`);
```

Comment:
Remove "for"?  Or just implement chunking ;)

Reply:
:-)